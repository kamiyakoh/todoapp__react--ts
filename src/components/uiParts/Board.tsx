import type { ReactNode } from 'react';
import type { SerializedStyles } from '@emotion/react';
/** @jsxRuntime classic */
/** @jsx jsx */
import { css, jsx } from '@emotion/react';
import { mq } from '../../styles/const';

interface Props {
  cssName?: SerializedStyles | SerializedStyles[];
  children: ReactNode;
}

export const Board = ({ cssName, children }: Props): JSX.Element => {
  return <div css={[board, cssName]}>{children}</div>;
};

const board = css`
  position: relative;
  padding: 1em;
  width: 100%;
  background: #006633;
  color: #fff;
  border: 8px solid #b2771f;
  border-radius: 3px;
  box-shadow: 0 0 5px #333, 0 0 5px #555 inset;
  ${mq('pc')} {
    width: calc(50% - 1em);
  }

  &:before,
  &:after {
    content: '';
    position: absolute;
    bottom: 0;
  }

  &:before {
    width: 20px;
    right: 10px;
    border: solid 3px var(--color, #6fd1ff);
    border-radius: 3px 2px 0 2px;
  }

  &:after {
    width: 15px;
    right: 45px;
    border: solid 3px #fff;
    border-radius: 8px 5px 2px 5px;
  }
`;
