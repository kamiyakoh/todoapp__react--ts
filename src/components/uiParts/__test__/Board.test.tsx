import { render } from '@testing-library/react';
import { Board } from '../Board';

describe('Board Test', () => {
  test('render', () => {
    const { asFragment } = render(<Board>children</Board>);
    expect(asFragment()).toMatchSnapshot();
  });
});
