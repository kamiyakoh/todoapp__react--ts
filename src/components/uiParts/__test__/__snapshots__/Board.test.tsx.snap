// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Test render 1`] = `
<DocumentFragment>
  <div
    class="css-1p9mkd4-Board"
  >
    children
  </div>
</DocumentFragment>
`;
